src/figuresClasses.ts

```diff
@@ -1,19 +1,82 @@
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Colors = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shapes,
+  color: Colors,
+  getArea(): number,
+}
 
+function validSides(...sides: number[]): void {
+  if (sides.some((side) => side <= 0)) {
+    throw new Error('Each side value should be bigger than 0!');
+  }
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = Shapes.Triangle;
 
-}
+  constructor(
+    public color: Colors,
+    public a: number = 0,
+    public b: number = 0,
+    public c: number = 0,
+  ) {
+    validSides(a, b, c);
+
+    if (a + b <= c || a + c <= b || c + b <= a) {
+      throw new Error('Invalid side values!');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
 
+    return Math.floor(area * 100) / 100;
```

Comment:
you can use destructuring for better code readability and reducing the amount of code
```suggestion
    const { a, b, c } = this;
    const s = (a + b + c) / 2;
    const area = Math.sqrt(s * (s - a) * (s - b) * (s - c));

    return Math.floor(area * 100) / 100;
```

Reply:
Good point